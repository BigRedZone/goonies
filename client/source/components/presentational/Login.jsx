import React from 'react';
import axios from 'axios';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleLogin() {
    const LoginInfo = {
      username: this.state.username,
      password: this.state.password,
    };

    axios.get('/api/users/login', {
      params: LoginInfo,
    })
      .then((res) => {
        if (res.data === 'invalid') {
          alert('Invalid username and password. Please try again.');
        } else {
          // success! redirect
        }
      })
      .catch((err) => {
        console.log('login error', err);
      });
  }


  render() {
    return (
      <div>
        <form>

          <label>
Username:
            <input
              type="text"
              name="username"
              onChange={this.handleChange}
            />
            <br />
          </label>

          <label>
Password:
            <input
              type="password"
              name="password"
              onChange={this.handleChange}
            />
            <br />
          </label>

          <input
            type="button"
            value="Login"
            onClick={this.handleLogin}
          />
        </form>
      </div>
    );
  }
}

export default Login;
