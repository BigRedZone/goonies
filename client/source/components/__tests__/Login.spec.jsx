import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../presentational/Login';

test('Login Renders correctly', () => {
  const component = renderer.create(<Login />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
