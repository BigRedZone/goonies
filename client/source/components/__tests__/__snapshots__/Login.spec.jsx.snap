// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders correctly 1`] = `
<div
  className="login"
>
  <form>
    <div
      className="loginfield"
    >
      <label
        htmlFor="username"
      >
        Username:
        <br />
        <input
          name="username"
          onChange={[Function]}
          type="text"
        />
      </label>
      <br />
      <label
        htmlFor="password"
      >
        Password:
        <br />
        <input
          name="password"
          onChange={[Function]}
          type="password"
        />
      </label>
      <br />
    </div>
    <div
      className="loginPageButtons"
    >
      <input
        className="login"
        onClick={[Function]}
        type="button"
        value="Login"
      />
      <input
        onClick={[Function]}
        type="button"
        value="New User? Register"
      />
    </div>
  </form>
</div>
`;
